syntax = "proto3";

package demo;

service DemoService {
    rpc sendData (DataPacket) returns (DataResponse);
}

message Data {
    double Decimal1 = 1;
    double Decimal2 = 2;
    double Decimal3 = 3;
    double Decimal4 = 4;
    string Timestamp = 5;
}

message DataPacket {
    string PacketTimestamp = 1;
    int32 PacketSeqNum = 2;
    int32 NRecords = 3;
    repeated Data PacketData = 4;
}

message DataResponse {
    string message = 1;
}
